speculos/main.py

```diff
@@ -317,6 +335,7 @@ def main(prog=None) -> int:
 
     # Init model and api_level if not specified from app elf metadata
     app_path = getattr(args, 'app.elf')
+    is_bagl = get_elf_fonts_size(app_path) != 0
```

Comment:
It's strictly impossible to have BAGL and NBGL calls at the same time, so it's not a big deal to know if NBGL or BAGL at instantiation time

Reply:
ah! What makes it impossible for an app to calls both NBGL and BAGL syscalls? The current implementation in the C SDK? Or something on the OS side?

Reply:
I'm also wondering of Blue and Nanos, as far as I known the fonts are handled by the MCU and therefore not in the app code, but they are still using BAGL? So this check might need to exclude them?

Reply:
In the SDK you have to choose between HAVE_BAGL and HAVE_NBGL, you cannot set both, and it wouldn't make any sense...

Reply:
Nano S and Blue should not be impacted by this PR, which only changes things for NBGL targets

Reply:
> In the SDK you have to choose between HAVE_BAGL and HAVE_NBGL, you cannot set both, and it wouldn't make any sense...

Could make sense to port on NBGL only part of the UX flows (transaction review) without bothering doing it for other UX flow more complex to adpat. Settings UX flow for example.
This is an edge case, but if we want Speculos to emulate correctly Bolos behavior I think this might be interesting to handle this better. Especially if we can find a not so dirtier solution to do it.

Reply:
> Nano S and Blue should not be impacted by this PR, which only changes things for NBGL targets

Then `is_bagl` should be set to True on these targets without looking for the fonts on the app elf?

Reply:
```
$objdump -s -t build/nanos/bin/app.elf | grep C_bagl
$objdump -s -t build/nanos2/bin/app.elf | grep C_bagl
c0de60f8 g     O .text  0000000c C_bagl_fonts
c0de6104 g     O .text  00000004 C_bagl_fonts_count
```

Reply:
it works when looking for the fonts on the app elf for all targets, so...

Reply:
> it works when looking for the fonts on the app elf for all targets, so...

You mean here: https://github.com/LedgerHQ/speculos/blob/master/speculos/main.py#L113 ?
Cause on NanoS apps `bagl_fonts_symbol` get the value `None` here, which would therefore get the size 0 from `get_elf_fonts_size` in your code, and therefore considering that NanoS use NBGL, which is not correct.
Did I missed something?

Reply:
You're right, I've wrongly read your comment, I will filter with HW model